src/main/kotlin/the/coding/force/exploring_kotlin_coroutines/service/coroutine/ReadDataServiceCoroutine.kt

```diff
@@ -8,12 +8,32 @@ import the.coding.force.exploring_kotlin_coroutines.exception.DataNotFoundExcept
 import the.coding.force.exploring_kotlin_coroutines.repository.DataRepository
 import the.coding.force.exploring_kotlin_coroutines.response.ReadDataResponse
 
+/**
+ * This Service has the suspend method `read` inside coroutine context to get a `ReadDataResponse`
+ *
+ * @author Lucas Terra
+ * @param dataRepository repository from DataEntity
+ * @property read suspend method to get a `ReadDataResponse` in coroutine context
+ */
 @Service
 class ReadDataServiceCoroutine(
     private val dataRepository: DataRepository,
 ) {
+    // logger to register the status of suspend method read
     private val logger = KotlinLogging.logger { }
 
+    /**
+     * received a `dataId` from suspend controller `read` and try to find the data to get it,
+     * if the data exists it will be got in the database, case the data does not exist an exception
+     * DataNotFoundException` will be thrown. The `logger` will register the status of read operation
+     *
+     * @author Lucas Terra
+     * @param dataId ID from `DataEntity` received from suspend controller `read` to find in the database
+     * @return `ReadDataResponse` case a register exists in the database
+     * @throws DataNotFoundException when the data with `DataId` was not found this exception is thrown
```

Comment:
```suggestion
    /**
     * @author Lucas Terra
     * @param dataId ID of the `DataEntity` to search
     * @return `ReadDataResponse` content of searched DataEntity if exists in the database
     * @throws DataNotFoundException when `DataEntity`was not found 
```

Cortei uns pedaços pra deixar mais conciso.